src/server/models/videos.js

```diff
@@ -1 +1,45 @@
-module.exports = {};
+const { collection, getDocs, doc, setDoc, updateDoc, deleteDoc } = require('firebase/firestore');
+
+const dbPromise = require('../../config/firebase');
+
+module.exports = {
+  getAllVideos: async () => {
+    try {
+      const db = await dbPromise;
+      const videoRef = collection(db, 'videos');
+      const querySnapshot = await getDocs(videoRef);
+      const documents = querySnapshot.docs.map((doc) => ({ id: doc.id, ...doc.data() }));
+      return documents;
+    } catch (err) {
+      console.error(err);
+    }
```

Comment:
Needs return err after console.error

Reply:
I'm gonna merge it and fix the errors, because it seems kevin has the updated db.js code